include/boost/math/tools/polynomial.hpp

```diff
@@ -455,6 +454,18 @@ class polynomial :
        normalize();
        return *this;
    }
+   
+   // Useful for fast test of equality to zero.
+   operator bool() const
```

Comment:
This should be explicit when BOOST_NO_CXX11_EXPLICIT_CONVERSION_OPERATORS is not set, otherwise use an implicit conversion to an "unmentionable" type, for example multiprecision::number uses:

   typedef bool (self_type::*unmentionable_type)()const;

   BOOST_MP_FORCEINLINE operator unmentionable_type()const
   {
      return is_zero() ? 0 : &self_type::is_zero;
   }

Which prevents accidental conversion to an arithmetic type.